### ISSUE TITLE
Kill page cache on preview/submit

### ISSUE THREAD

**Issue by gerritvanaaken** [Addpixel/KirbyComments#13]
If you have kirby file cache activated, you will never see the comment preview, and so you can never send any comment :-( Maybe there is a kirby api for explicitely deleting the cache for the page you are commenting on.

**Comment by florianpircher** [comment]
There is: [`$page->reset()`](https://getkirby.com/docs/cheatsheet/page/reset). The only problem with this method is, that it never will be called, because the page is loaded from the cache and so the method is not getting called. I try to look for an alternative …

**Comment by florianpircher** [comment]
And adding the current page temporarily to [`cache.ignore`](https://getkirby.com/docs/cheatsheet/options/cache-ignore) does not work, because pages are cached before the code of plugin is executed.

**Comment by florianpircher** [comment]
It seems like this problem can currently not be fixed. Comments are dynamic content and don’t go together with Kirby’s caching strategy.

**Comment by gerritvanaaken** [comment]
Maybe the Kirby team has to change the caching bevahiour. If a page is called with a post request and post data incoming, the cache will not be used.

**Comment by florianpircher** [comment]
I found a black-magic method of ignoring the cache when previewing or posting a comment. Now I have discovered, that you would also have to ignore the page when rendering the form, regardless of whether or not a comment was posted. The form contains a hidden input field with a random key for extra security (e.g. to prevent cross-site request forgery). This means, that you would have to ignore your articles at all times from the cache.

This can be achieved using a wildcard ignore pattern:

```php
c::set('cache.ignore', array(
  'weblog/*'
));
```
